x-pack/plugins/upgrade_assistant/common/constants.ts

```diff
@@ -17,7 +17,7 @@ export const MAJOR_VERSION = '7.16.0';
  */
 export const indexSettingDeprecations = {
   translog: {
-    deprecationMessage: 'translog retention settings are ignored', // expected message from ES deprecation info API
+    deprecationMessage: 'Translog retention settings are deprecated', // expected message from ES deprecation info API
```

Comment:
@alisonelizabeth Is this bug caused because we're depending upon an exact string match with the deprecation message returned by the Deprecation Info API?

Reply:
Similarly, the ML upgrade process also depends on reading strings from the deprecation message making it quite fragile to changes. A [similar bug](https://github.com/elastic/kibana/pull/122541) occurred few weeks ago, that turned out to be due to changes made deprecation message.

Reply:
That's correct. As mentioned in the PR description, this is a similar problem fixed via https://github.com/elastic/kibana/pull/119745.

ES did add the ability to include metadata in the deprecation info API via https://github.com/elastic/elasticsearch/issues/73089 that we could potentially leverage in the future. Given the time constraints though, for now, I have patched this and added an API test to catch future regressions. There's some more discussion around this on https://github.com/elastic/kibana/issues/119834.

Reply:
@elastic/es-data-management Just raising awareness of this brittle dependency that the UI has upon the API.